cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -51,10 +51,9 @@ public class InMemorySourceControlOperationRunner implements SourceControlOperat
   }
 
   @Override
-  public PushAppResponse push(NamespaceId namespace, RepositoryConfig repoConfig,
-                              ApplicationDetail appToPush, CommitMeta commitDetails)
-    throws PushFailureException, NoChangesToPushException, AuthenticationConfigException {
-    RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig);
+  public PushAppResponse push(NamespaceId namespace, ApplicationDetail appToPush, CommitMeta commitDetails)
+    throws PushFailureException, NoChangesToPushException, AuthenticationConfigException, RepositoryNotFoundException {
+    RepositoryManager repositoryManager = repoManagerFactory.create(namespace);
```

Comment:
You should just use try with resource instead of try/finally.

Reply:
yeah we tried to use `try with resource` pattern like below:

```
try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace)) {
      try {
        repositoryManager.cloneRemote();
      } catch (GitAPIException | IOException e) {
        throw new PushFailureException(String.format("Failed to clone remote repository: %s", e.getMessage()), e);
      }

      LOG.info("Pushing application configs for : {}", appToPush);

      //TODO: CDAP-20371, Add retry logic here in case the head at remote moved while we are doing push
      return writeAppDetailAndPush(repositoryManager, appToPush, commitDetails);
    } catch (IOException e) {
      LOG.warn("Failed to close the RepositoryManager, there may be leftover files", e);
    }
```

It's just that this will end up missing return. The issue is caused by possible `IOException` from `RepositoryManager#close()`. If it fails to close, we still should return the pushed result and only log the IO exception error. fyi, @albertshau

Reply:
Moved the IOException log in RepoManager now.